module/x/gravity/abci_test.go

```diff
@@ -34,6 +34,35 @@ func TestValsetCreationIfNotAvailable(t *testing.T) {
 	require.True(t, len(valsets) == 1)
 }
 
+func TestValsetShutdown(t *testing.T) {
+	input, ctx := keeper.SetupFiveValChain(t)
+	defer func() { input.Context.Logger().Info("Asserting invariants at test end"); input.AssertInvariants() }()
+	pk := input.GravityKeeper
+
+	// EndBlocker should set a new validator set if not available
+	EndBlocker(ctx, pk)
+	require.NotNil(t, pk.GetValset(ctx, uint64(pk.GetLatestValsetNonce(ctx))))
+	valsets := pk.GetValsets(ctx)
+	require.True(t, len(valsets) == 1)
+
+	// enable shutdown, and check that we created a new valset with a single member (zero address)
+	pk.Shutdown()
+	EndBlocker(ctx, pk)
+	require.NotNil(t, pk.GetValset(ctx, uint64(pk.GetLatestValsetNonce(ctx))))
```

Comment:
let's also check that the latest valse is correct, rather than only checking that it is not nil.

Reply:
There's a check for it below:

```go
require.Len(t, valsets[0].Members, 1)
require.Equal(t, valsets[0].Members[0].EthereumAddress, types.ZeroAddress().GetAddress().Hex())
```